C7/Civ3Unit/Civ3Unit.cs

```diff
@@ -2,177 +2,173 @@
+
+		int bmpSize = width * height * 4 + 54;  //54 = Windows 3 BMP header size.
+		byte[] bmpBuffer = new byte[bmpSize];
+		fixed (byte* fPtr = bmpBuffer) {
+			//BMP header.  This is a mix of byte, short, and int.  Probably a cleaner way to do this, but my C is rusty.
+			byte* bPtr = fPtr;
+			int* iPtr;
+			short* sPtr;
+			bPtr[0] = 66;    //B
+			bPtr[1] = 77;    //M
+			bPtr+=2;
+			iPtr = (int*)bPtr;
+			iPtr[0] = bmpSize;  //size of BMP file in bytes
+			iPtr[1] = 0;    //reserved, two shorts
+			iPtr[2] = 54;   //offset of image data from start of file
+			iPtr[3] = 40;   //size of Windows 3 BMP header
+			iPtr[4] = width;
+			iPtr[5] = height;
+			bPtr+=24;   //6 * 4
+			sPtr = (short*)bPtr;
+			sPtr[0] = 1;    //num color planes
+			sPtr[1] = 32;   //bit depth.  we want 32-bit with alpha
+			bPtr+=4;
+			iPtr = (int*)bPtr;
+			iPtr[0] = 0;    //compression - none.  Godot doesn't support compression
+			iPtr[1] = bmpSize - 54; //size, without headers
+			iPtr[2] = 0;    //horizontal resolution.  ignored.
+			iPtr[3] = 0;    //vertical resolution.  ignored.
+			iPtr[4] = 0;    //num colors in palette.  not relevant for 32-bit
+			iPtr[5] = 0;    //num important colors in palette.  not relevant.
+			bPtr+=24;   //6 * 4
+			iPtr = (int*)bPtr;
+			//Ready for image data
+
+			int c = 0;
+			//The BMP data is stored bottom-to-top, whereas our data is top-to-bottom
+			//Thus we'll use c to calculate the index for each row
+			for (int y = height - 1; y > -1; y--)
+			{
+				c = width * y;
+				for (int x = 0; x < width; x++)
+				{
+					if (shadows && colorIndices[c] > 239) {
+						// using black and transparency
+						iPtr[0] = ((255 - colorIndices[c]) << 4) << 24;
+						// using the palette color but adding transparency
+						// OutImage.SetPixel(i % width, i / width, Color.Color8(palette[ba[i],0], palette[ba[i],1], palette[ba[i],2], (byte)((255 -ba[i]) * 16)));
+					} else {
+						// A, R, G, B
+						iPtr[0] = ((byte)255) << 24 | palette[colorIndices[c], 0] << 16 | palette[colorIndices[c], 1] << 8 | palette[colorIndices[c], 2] << 0;
+					}
+					c++;
+					iPtr++;
+				}
+			}
+		}
+		return bmpBuffer;
+	}
 }
-// This is just to add some movement to an AnimatedSprite
-public class MovingSprite : AnimatedSprite {
-    public Vector2 Velocity = new Vector2(0, 0);
-    public override void _PhysicsProcess(float delta) {
-        Position = Position + Velocity;
-        if (Position.x > 1040) {
-            Position = new Vector2(-30, Position.y);
-        }
-        if (Position.x < -30) {
-            Position = new Vector2(1040, Position.y);
-        }
-        if (Position.y > 800) {
-            Position = new Vector2(Position.x, -30);
-        }
-        if (Position.y < -30) {
-            Position = new Vector2(Position.x, 800);
-        }
-    }
+// This is just to add some movement to an AnimatedSprite2D
+public partial class MovingSprite : AnimatedSprite2D {
```

Comment:
what does this movement do and where do the numbers come from? is it assuming something about window size?